lib/ansible/module_utils/network/ftd/device.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2019 Cisco and/or its affiliates.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from ansible.module_utils.six.moves.urllib.parse import urlparse
+
+try:
+    from kick.device2.ftd5500x.actions.ftd5500x import Ftd5500x
+    from kick.device2.kp.actions import Kp
+
+    HAS_KICK = True
+except ImportError:
+    HAS_KICK = False
+
+
+class FtdModel:
+    FTD_ASA5506_X = 'Cisco ASA5506-X Threat Defense'
+    FTD_ASA5508_X = 'Cisco ASA5508-X Threat Defense'
+    FTD_ASA5516_X = 'Cisco ASA5516-X Threat Defense'
+
+    FTD_2110 = 'Cisco Firepower 2110 Threat Defense'
+    FTD_2120 = 'Cisco Firepower 2120 Threat Defense'
+    FTD_2130 = 'Cisco Firepower 2130 Threat Defense'
+    FTD_2140 = 'Cisco Firepower 2140 Threat Defense'
```

Comment:
Does this support 4100s and 9000 series appliances?

Reply:
Not as of now. We have tested the `ftd_install` module to work correctly with 2100/5500 series appliances so far. 4100 and 9000 are next on our backlog list, but their support must be added in a separate PR.